eng/ci/templates/jobs/build-artifacts.yml

```diff
@@ -2,14 +2,16 @@ parameters:
   - name: SkipBuildArtifacts
     type: boolean
     default: false
+  - name: PoolName
+    type: string
 
 jobs:
 
 - job: BuildArtifacts
   displayName: Build Artifacts
 
   pool:
-    name: 1es-pool-azfunc
+    name: ${{ parameters.PoolName }}
```

Comment:
Another approach would be to remove this `pool` section entirely. With it removed, it will inherit the pool specified at the root level (public-build.yml and official-build.yml)